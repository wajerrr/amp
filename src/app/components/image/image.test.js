import React from 'react';
import renderer from 'react-test-renderer';
import Image from './image';

describe('Image', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <Image
          src="https://www.economist.com/sites/default/files/20180602_blp904.jpg"
          width={100}
          height={100}
          layout="responsive"
        >
          link
        </Image>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
