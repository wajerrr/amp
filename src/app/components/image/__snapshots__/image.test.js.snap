// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should match snapshot 1`] = `
<amp-img
  alt=""
  height={100}
  layout="responsive"
  sizes="(min-width: 640px) calc(640px -  2 * 0.5em), calc(100vw - 2 * 0.5em)"
  src="https://www.economist.com/sites/default/files/20180602_blp904.jpg"
  srcset="https://www.economist.com/sites/default/files/imagecache/200-width/sites/default/files/20180602_blp904.jpg 200w,
https://www.economist.com/sites/default/files/imagecache/300-width/sites/default/files/20180602_blp904.jpg 300w,
https://www.economist.com/sites/default/files/imagecache/400-width/sites/default/files/20180602_blp904.jpg 400w,
https://www.economist.com/sites/default/files/imagecache/640-width/sites/default/files/20180602_blp904.jpg 640w,
https://www.economist.com/sites/default/files/imagecache/800-width/sites/default/files/20180602_blp904.jpg 800w,
https://www.economist.com/sites/default/files/imagecache/1000-width/sites/default/files/20180602_blp904.jpg 1000w,
https://www.economist.com/sites/default/files/imagecache/1200-width/sites/default/files/20180602_blp904.jpg 1200w,
https://www.economist.com/sites/default/files/imagecache/1280-width/sites/default/files/20180602_blp904.jpg 1280w,
https://www.economist.com/sites/default/files/imagecache/1600-width/sites/default/files/20180602_blp904.jpg 1600w"
  width={100}
/>
`;

exports[`Image should match snapshot with image src outside www.economist.com domain 1`] = `
<amp-img
  alt=""
  height={100}
  layout="responsive"
  sizes={undefined}
  src="https://www.theguardian.com/sites/default/files/20180602_blp904.jpg"
  srcset={undefined}
  width={100}
/>
`;
